data-prepper-api/src/main/java/com/amazon/dataprepper/model/metric/ParameterValidator.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.model.metric;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+public class ParameterValidator {
```

Comment:
It seems this class is only required for the `Jackson` classes in this package. Please make it package protected so that it does not get used beyond there. In Java, you just remove the `public` modifier before class.

```
class ParameterValidator
```